File: app/src/main/java/com/coder/ffmpegtest/ui/dialog/PromptDialog.java
package com.coder.ffmpegtest.ui.dialog;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.Gravity;
import android.view.KeyEvent;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.coder.ffmpegtest.R;
import com.coder.ffmpegtest.base.BaseDialog;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.BindView;
import butterknife.OnClick;

public class PromptDialog extends BaseDialog {

    @BindView(R.id.tv_title)
    TextView tvTitle;
    @BindView(R.id.tv_content)
    TextView tvContent;
    @BindView(R.id.btn_positive)
    Button btnPositive;
    @BindView(R.id.btn_negative)
    Button btnNegative;
    @BindView(R.id.vi_line)
    View line;
    @BindView(R.id.progress_bar)
    ProgressBar mProgressBar;

    private int gravity = Gravity.LEFT;
    private boolean hasNegativeButton = true;
    private boolean canceledOutside = true;


    public static PromptDialog newInstance(@Nullable String title, @NonNull String content,
                                           @Nullable String negative, @Nullable String positive) {
        PromptDialog fragment = new PromptDialog();
        Bundle args = new Bundle();
        args.putString("title", title);
        args.putString("content", content);
        args.putString("negative", negative);
        args.putString("positive", positive);
        fragment.setArguments(args);
        return fragment;
    }

    private OnPromptListener mOnPromptListener;

    public void setContentGravity(int gravity) {
        this.gravity = gravity;
    }

    @Override
    public int getLayoutId() {
        return R.layout.dialog_prompt;
    }

    @Override
    protected void init() {
        initData();
    }

    private void initData() {
        Bundle args = getArguments();

        setTitle(args.getString("title"));
        setNegativeStr(args.getString("negative"));
        setPositiveStr(args.getString("positive"));

        tvContent.setGravity(gravity);

        btnNegative.setVisibility(hasNegativeButton ? View.VISIBLE : View.GONE);
        line.setVisibility(hasNegativeButton ? View.VISIBLE : View.GONE);
        setCanceledOnTouchOutside(canceledOutside);
    }

    public void setHasNegativeButton(boolean hasNegativeButton) {
        this.hasNegativeButton = hasNegativeButton;
    }

    @OnClick({R.id.btn_positive, R.id.btn_negative})
    public void onClick(View view) {
        if (view.getId() ==R.id.btn_positive ){
            if (mOnPromptListener != null) {
                mOnPromptListener.onPrompt(true);
            }
        }else {
            if (mOnPromptListener != null) {
                mOnPromptListener.onPrompt(false);
            }
        }
        dismissAllowingStateLoss();
    }

    public void setOnPromptListener(OnPromptListener onPromptListener) {
        mOnPromptListener = onPromptListener;
    }

    public void setCanceledOutside(boolean canceledOutside) {
        this.canceledOutside = canceledOutside;
    }

    public void setTitle(String title) {
        if (!TextUtils.isEmpty(title)) {
            tvTitle.setVisibility(View.VISIBLE);
            tvTitle.setText(title);
        } else {
            tvTitle.setVisibility(View.GONE);
        }
    }

    public void setContent(int progress) {
        if (tvContent!=null){
            tvContent.setText(String.format("进度: %s", progress));
        }
       if (mProgressBar!=null){
           mProgressBar.setProgress(progress);
       }

    }

    private void setPositiveStr(String positiveStr) {
        if (!TextUtils.isEmpty(positiveStr)) {
            btnPositive.setText(positiveStr);
        }
    }

    private void setNegativeStr(String negativeStr) {
        if (!TextUtils.isEmpty(negativeStr)) {
            btnNegative.setText(negativeStr);
        }
    }

    public interface OnPromptListener {
        /**
         * 返回用户点击【确定】/【取消】
         *
         * @param isPositive 是否点击确定
         */
        void onPrompt(boolean isPositive);
    }

    @Override
    public void onResume() {
        super.onResume();
        getView().setFocusableInTouchMode(true);
        getView().requestFocus();
        getView().setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View view, int i, KeyEvent keyEvent) {
                if(keyEvent.getAction() == KeyEvent.ACTION_DOWN && i == KeyEvent.KEYCODE_BACK){
                    return true;
                }
                return false;
            }
        });
    }
}

File: app/src/main/java/com/coder/ffmpegcommand/ui/dialog/PromptDialog.kt
package com.coder.ffmpegcommand.ui.dialog

import android.annotation.SuppressLint
import android.os.Bundle
import android.text.TextUtils
import android.view.Gravity
import android.view.KeyEvent
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import android.widget.TextView
import com.coder.ffmpegcommand.R
import com.coder.ffmpegcommand.base.BaseDialog

@SuppressLint("NonConstantResourceId")
class PromptDialog : BaseDialog(),View.OnClickListener {
    var tvTitle: TextView? = null
    var tvContent: TextView? = null
    var btnPositive: Button? = null
    var btnNegative: Button? = null
    var line: View? = null
    var mProgressBar: ProgressBar? = null

    private var gravity = Gravity.LEFT
    private var hasNegativeButton = true
    private var canceledOutside = true
    private var mOnPromptListener: OnPromptListener? = null
    fun setContentGravity(gravity: Int) {
        this.gravity = gravity
    }

    override val layoutId: Int
        get() = R.layout.dialog_prompt

    override fun init() {
        initView()
        initData()
    }

    private fun initView(){
        tvTitle = findViewById(R.id.tv_title) as TextView?
        tvContent = findViewById(R.id.tv_content) as TextView?
        btnPositive = findViewById(R.id.btn_positive) as Button?
        btnNegative = findViewById(R.id.btn_negative) as Button?
        line = findViewById(R.id.vi_line)
        mProgressBar = findViewById(R.id.progress_bar) as ProgressBar?

        btnNegative?.setOnClickListener(this)
        btnPositive?.setOnClickListener(this)
    }

    private fun initData() {
        val args = arguments
        setTitle(args!!.getString("title"))
        setNegativeStr(args.getString("negative"))
        setPositiveStr(args.getString("positive"))
        tvContent!!.gravity = gravity
        btnNegative!!.visibility = if (hasNegativeButton) View.VISIBLE else View.GONE
        line!!.visibility = if (hasNegativeButton) View.VISIBLE else View.GONE
        setCanceledOnTouchOutside(canceledOutside)
    }

    fun setHasNegativeButton(hasNegativeButton: Boolean) {
        this.hasNegativeButton = hasNegativeButton
    }


    override fun onClick(view: View) {
        if (view.id == R.id.btn_positive) {
            if (mOnPromptListener != null) {
                mOnPromptListener!!.onPrompt(true)
            }
        } else {
            if (mOnPromptListener != null) {
                mOnPromptListener!!.onPrompt(false)
            }
        }
        dismissAllowingStateLoss()
    }

    fun setOnPromptListener(onPromptListener: OnPromptListener?) {
        mOnPromptListener = onPromptListener
    }

    fun setCanceledOutside(canceledOutside: Boolean) {
        this.canceledOutside = canceledOutside
    }

    fun setTitle(title: String?) {
        if (!TextUtils.isEmpty(title)) {
            tvTitle?.visibility = View.VISIBLE
            tvTitle?.text = title
        } else {
            tvTitle?.visibility = View.GONE
        }
    }

    fun setContent(progress: Int) {
        if (tvContent != null) {
            tvContent?.text = String.format("进度: %s", progress)
        }
        if (mProgressBar != null) {
            mProgressBar?.progress = progress
        }
    }

    private fun setPositiveStr(positiveStr: String?) {
        if (!TextUtils.isEmpty(positiveStr)) {
            btnPositive?.text = positiveStr
        }
    }

    private fun setNegativeStr(negativeStr: String?) {
        if (!TextUtils.isEmpty(negativeStr)) {
            btnNegative?.text = negativeStr
        }
    }

    interface OnPromptListener {
        /**
         * 返回用户点击【确定】/【取消】
         *
         * @param isPositive 是否点击确定
         */
        fun onPrompt(isPositive: Boolean)
    }

    override fun onResume() {
        super.onResume()
        view!!.isFocusableInTouchMode = true
        view!!.requestFocus()
        view!!.setOnKeyListener { view, i, keyEvent ->
            if (keyEvent.action == KeyEvent.ACTION_DOWN && i == KeyEvent.KEYCODE_BACK) {
                true
            } else false
        }
    }

    companion object {
        fun newInstance(title: String?, content: String,
                        negative: String?, positive: String?): PromptDialog {
            val fragment = PromptDialog()
            val args = Bundle()
            args.putString("title", title)
            args.putString("content", content)
            args.putString("negative", negative)
            args.putString("positive", positive)
            fragment.arguments = args
            return fragment
        }
    }
}